### ISSUE TITLE
Nested pkg update not working 

### ISSUE THREAD

**Issue by bharathkkb** [GoogleContainerTools/kpt#2379]
I have composed a pkg like so
```
kptv1 pkg get https://github.com/bharathkkb/blueprints.git/catalog/bucket@kpt-v1 top-bucket
kptv1 pkg get https://github.com/bharathkkb/blueprints.git/catalog/bucket@kpt-v1 top-bucket/nested-bucket
```
I wanted to update both packages to a newer version of the upstream bucket pkg which exists on a branch
```
kptv1 pkg update top-bucket@feat/bucket-new
```
CLI output shows
```
Package "top-bucket":
Fetching upstream from https://github.com/bharathkkb/blueprints@feat/bucket-new
From https://github.com/bharathkkb/blueprints
 * branch            feat/bucket-new -> FETCH_HEAD
Fetching origin from https://github.com/bharathkkb/blueprints@feat/bucket-new
From https://github.com/bharathkkb/blueprints
 * branch            4bb5f7fcf35f750c15eac4f0ed8c4170a2474369 -> FETCH_HEAD
Updating package "top-bucket" with strategy "resource-merge".

Package "top-bucket/nested-bucket":
Fetching upstream from https://github.com/bharathkkb/blueprints@kpt-v1
From https://github.com/bharathkkb/blueprints
 * branch            kpt-v1     -> FETCH_HEAD
Fetching origin from https://github.com/bharathkkb/blueprints@kpt-v1
From https://github.com/bharathkkb/blueprints
 * branch            4bb5f7fcf35f750c15eac4f0ed8c4170a2474369 -> FETCH_HEAD
Updating package "nested-bucket" with strategy "resource-merge".

Updated 2 package(s).
```

However only the parent package (top-bucket) is updated although CLI op seems to show 2 were updated
```
git diff --name-only
top-bucket/Kptfile
top-bucket/bucket.yaml
```

### Expected behavior

I expected both packages to be updated to the specified version. 

Kptfile(s) in all subpkgs should be edited to the specified version via CLI + updated

### Actual behavior

Only parent package was updated to the specified version. Subpkgs were updated but not to the version specified via CLI.

### Information

Version: 1.0.0-beta.1

### Steps to reproduce the behavior

**Comment by phanimarupaka** [comment]
@bharathkkb This is expected behavior. Since you are invoking update imperatively by specifying the version, the new version is only set in the parent package Kptfile and update is triggered. This means the parent package should be updated to `feat/bucket-new` version and child should be updated to `kpt-v1` version(which it already is on). The right way to update such nested packages is to manually edit the `ref` in  each `Kptfile` and invoke `kpt pkg update top-bucket`.

**Comment by phanimarupaka** [comment]
Related issue https://github.com/GoogleContainerTools/kpt/issues/1941

**Comment by bharathkkb** [comment]
@phanimarupaka per go/update-subpackages implementation it seemed to imply that the subpkgs would be updated recursively. If this is not the case, can we consider this as a FR?

**Comment by phanimarupaka** [comment]
@bharathkkb Per https://kpt.dev/book/03-packages/05-updating-a-package?id=update-the-package specifying the version in CLI command is just a `porcelain to editing Kptfile + kpt pkg update`. The specified version in the CLI command only updates the parent Kptfile and not the children. So it seems to you that the packages are not being updated recursively. But the fact is they are being updated, however the child package is not being provided the version to update from CLI. 

In short, if you want to update packages recursively, edit each Kptfile in all packages with desired version and trigger `kpt pkg update` once on the parent package.

**Comment by morgante** [comment]
@phanimarupaka We are asking this to be a FR. Manually editing each Kptfile in all packages to the desired version is very toilsome. /cc @mikebz

**Comment by phanimarupaka** [comment]
@morgante If this is a feature request, can you update the description with the requirements clearly.

**Comment by bharathkkb** [comment]
@phanimarupaka I have tweaked the description. The ask is basically to recursively update the refs in subpkgs kptfiles to desired version as specified via CLI followed by `kpt pkg update`.
